fairscale/optim/oss.py

```diff
@@ -125,29 +137,12 @@ def partition_parameters(self) -> List[List[dict]]:
         inside step().
         """
         if len(self._partition_parameters) == 0:
-            self._partition_parameters = [list() for _ in range(self.world_size)]
-            sizes = [0] * self.world_size
-            for param_group in self.param_groups:
-                param_lists: List[List] = [list() for _ in range(self.world_size)]
-                for param in param_group["params"]:
-                    # Add this param to rank with smallest size.
-                    rank = sizes.index(min(sizes))
-                    param_lists[rank].append(param)
-
-                    # We're partitioning the optimizer state,
-                    # so trainable parameters are the ones which really count
-                    if param.requires_grad:
-                        sizes[rank] += param.numel()
-                    else:
-                        # Spread frozen params on a per-tensor basis
-                        # Mostly useful for balance partitions for fine tuning for instance
-                        # Not required strictly speaking
-                        sizes[rank] += 1
-
-                for rank, params in enumerate(param_lists):
-                    param_group_rank = copy.copy(param_group)
-                    param_group_rank["params"] = params
-                    self._partition_parameters[rank].append(param_group_rank)
+            if self._partition_type == OSS.PartitionType.MEMORY:
+                self._partition_parameters = partition_memory(self.param_groups, self.world_size)
+
+            elif self._partition_type == OSS.PartitionType.ORDERING:
+                # TODO: update the partitioning and ordering given the backward pass information
```

Comment:
a follow up PR does the non-blocking-backward change from shardedDDP, and dynamic partitioning given the ordering learnt during the backward would be the final nail in the coffin